### PR TITLE
e2e_apimachinery:stop using deprecated framework.ExpectError

### PR DESCRIPTION
**Pull request by liyuerich**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

test/e2e/apimachinery  stop using deprecated framework.ExpectError

#### Which issue(s) this PR fixes:

Ref # https://github.com/kubernetes/kubernetes/pull/115961

#### Special notes for your reviewer:

NONE

#### Does this PR introduce a user-facing change?

```
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```
NONE
```

### CODE REVIEW

File: test/e2e/apimachinery/crd_validation_rules.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	"strings"
 
 	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
```
Comment (carlory) [inline test/e2e/apimachinery/crd_validation_rules.go:24]: please leave a blank line.

File: test/e2e/apimachinery/namespace.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ import (
 	admissionapi "k8s.io/pod-security-admission/api"
 
 	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
 	"k8s.io/apimachinery/pkg/types"
```
Comment (carlory) [inline test/e2e/apimachinery/namespace.go:45]: and here.

**Comment by liyuerich** [comment]
thanks @wojtek-t   and @carlory

**Pull request merged**